File: server/models/group.js
const mongoose = require('mongoose');

const groupSchema = mongoose.Schema({
    groupName: {
        required: true,
        type: String,
        trim: true,
    },
    groupDescription: {
        required: true,
        type: String,
        trim: true,
    },
    budget: {
        required: true,
        type: String,
    },
    admin: {
        required: false,
        type: String,
    },
    members: {
        required: false,
        type: Array,
    },
    groupExpenses: {
        required: false,
        type: [
            {
                _id: {
                    type: String,
                    required: true,
                },
                expenseDate: {
                    type: String,
                    required: true,
                },
                expenseAmount: {
                    type: Number,
                    required: true,
                },
                expenseDescription: {
                    type: String,
                    required: true,
                },
                spendorName: {
                    type: String,
                    required: true,
                },
                spendorEmail: {
                    type: String,
                    required: true,
                },
                spenderGroup: {
                    type: String,
                    required: false,
                },
            },
        ],
    },
});

const Group = mongoose.model('Group', groupSchema);
module.exports = Group;

File: lib/features/group/domain/entity/group.dart
class GroupEntity {
  String? id;
  String groupName;
  String groupDescription;
  String budget;
  String admin;
  List<dynamic> members;
  List<Map<String, dynamic>>? groupExpenses; // Updated type

  GroupEntity({
    this.id,
    required this.groupName,
    required this.groupDescription,
    required this.budget,
    required this.admin,
    required this.members,
    this.groupExpenses, // Optional
  });

  GroupEntity copyWith({
    String? id,
    String? groupName,
    String? groupDescription,
    String? budget,
    String? admin,
    List<dynamic>? members,
    List<Map<String, dynamic>>? groupExpenses, // Consistent type
  }) {
    return GroupEntity(
      id: id ?? this.id,
      groupName: groupName ?? this.groupName,
      groupDescription: groupDescription ?? this.groupDescription,
      budget: budget ?? this.budget,
      admin: admin ?? this.admin,
      members: members ?? this.members,
      groupExpenses: groupExpenses ?? this.groupExpenses, // Fixed assignment
    );
  }
}
